lib/gui/app/components/drive-selector/DriveSelectorModal.jsx

```diff
@@ -0,0 +1,320 @@
+const React = require('react')
+const { Modal } = require('rendition')
+const {
+  isDriveValid,
+  getDriveImageCompatibilityStatuses,
+  hasListDriveImageCompatibilityStatus,
+  COMPATIBILITY_STATUS_TYPES
+} = require('../../../../shared/drive-constraints')
+const store = require('../../models/store')
+const analytics = require('../../modules/analytics')
+const availableDrives = require('../../models/available-drives')
+const selectionState = require('../../models/selection-state')
+const { bytesToClosestUnit } = require('../../../../shared/units')
+const utils = require('../../../../shared/utils')
+const { open: openExternal } = require('../../os/open-external/services/open-external')
+const ConfirmModal = require('../confirm-modal/ConfirmModal.jsx')
+
+const DriveSelectorModal = (props) => {
+  const { close } = props
+  const [ confirmModal, setConfirmModal ] = React.useState({ open: false })
+
+  /**
+   * @summary Determine if we can change a drive's selection state
+   * @function
+   * @private
+   *
+   * @param {Object} drive - drive
+   * @returns {Promise}
+   *
+   * @example
+   * shouldChangeDriveSelectionState(drive)
+   *    .then((shouldChangeDriveSelectionState) => {
+   *        if (shouldChangeDriveSelectionState) doSomething();
+   *    });
+   */
+  const shouldChangeDriveSelectionState = (drive) => {
+    return isDriveValid(drive, selectionState.getImage())
+  }
+
+  /**
+   * @summary Toggle a drive selection
+   * @function
+   * @public
+   *
+   * @param {Object} drive - drive
+   * @returns {void}
+   *
+   * @example
+   * toggleDrive({
+   *   device: '/dev/disk2',
+   *   size: 999999999,
+   *   name: 'Cruzer USB drive'
+   * });
+   */
+  const toggleDrive = (drive) => {
+    const canChangeDriveSelectionState = shouldChangeDriveSelectionState(drive)
+
+    if (canChangeDriveSelectionState) {
+      analytics.logEvent('Toggle drive', {
+        drive,
+        previouslySelected: selectionState.isCurrentDrive(availableDrives.device),
+        applicationSessionUuid: store.getState().toJS().applicationSessionUuid,
+        flashingWorkflowUuid: store.getState().toJS().flashingWorkflowUuid
+      })
+
+      selectionState.toggleDrive(drive.device)
+    }
+  }
+
+  /**
+   * @summary Prompt the user to install missing usbboot drivers
+   * @function
+   * @public
+   *
+   * @param {Object} drive - drive
+   * @returns {void}
+   *
+   * @example
+   * installMissingDrivers({
+   *   linkTitle: 'Go to example.com',
+   *   linkMessage: 'Examples are great, right?',
+   *   linkCTA: 'Call To Action',
+   *   link: 'https://example.com'
+   * });
+   */
+  const installMissingDrivers = (drive) => {
+    if (drive.link) {
+      analytics.logEvent('Open driver link modal', {
+        url: drive.link,
+        applicationSessionUuid: store.getState().toJS().applicationSessionUuid,
+        flashingWorkflowUuid: store.getState().toJS().flashingWorkflowUuid
+      })
+
+      setConfirmModal({
```

Comment:
Since these are the only possible settings for the Confirm modal in this component, how about specifying all of the props directly on the modal in JSX, and only having `[showConfirmModal, setShowConfirmModal] =React.useState(false)` in state instead?

Reply:
Definitely, after refactoring the confirm modal a bit it makes more sense, it's still useful to have a `confirmModal` object though so we just spread it with (e.g.) `{...confirmModal.options}` and avoid having multiple variables around

Reply:
If there were multiple places where this was used, it does make sense, but it doesn't really make sense as it is now. Even then, I would rather have 2 modals with different props, rather than using an object, as it is easier to follow what is logic and what is presentation

Reply:
The thing is that the values are still dynamic so we would either need to set/unset each of them every time, as opposed to just having the simplified version of the confirm-modal (which is gone with these changes) like:
```javascript
setConfirmModal({
        open: true,
        options: {
          width: 400,
          title: drive.linkTitle,
          cancel: () => setConfirmModal({ open: false }),
          done: async (shouldContinue) => {
            try {
              if (shouldContinue) {
                openExternal(drive.link)
              } else {
                setConfirmModal({ open: false })
              }
            } catch (error) {
              analytics.logException(error)
            }
          },
          action: 'Yes, continue',
          cancelButtonProps: {
            children: 'Cancel'
          },
          children: drive.linkMessage || `Etcher will open ${drive.link} in your browser`
        }
      })
```

Reply:
They're not 2+ modals, it's the same modal with different parameters. While not all of them are dynamic (e.g. action, width, cancel and cancelButtonProps are not), it still leaves us with 3 variables just for a modal... It's not the end of the world of course, but I'd rather have just an object that we can possibly reuse in the future as it can easily be extracted